tools/dds/dds-sniffer/dds-sniffer.cpp

```diff
@@ -0,0 +1,280 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#include "dds-sniffer.hpp"
+
+#include <thread>
+
+#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
+#include <fastdds/dds/domain/qos/DomainParticipantQos.hpp>
+#include <fastdds/dds/subscriber/Subscriber.hpp>
+#include <fastdds/dds/subscriber/SampleInfo.hpp>
+#include <fastrtps/types/DynamicDataHelper.hpp>
+#include <fastrtps/types/DynamicDataFactory.h>
+
+#include <fastrtps/transport/UDPv4TransportDescriptor.h>
+
+
+int main(int argc, char** argv)
+{
+    eprosima::fastdds::dds::DomainId_t domain = 0;
+    uint32_t seconds = 0;
+
+    for(int i = 2; i <= argc; ++i)
+    {
+        if (strcmp(argv[i - 1], "--help") == 0 ||
+            strcmp(argv[i - 1], "-h") == 0)
+        {
+            std::cout << "dds-sniffer tool\n"
+                      << "\t-h --help: shows this menu\n" 
+                      << "\t-d --domain <ID>: select domain ID to listen on\n" 
+                      << "\t-s --snapshot: run momentarily taking a snapshot of the domain\n" << std::endl;
+
+            return 0;
+        }
+
+        if (strcmp(argv[i - 1], "--snapshot") == 0 ||
+            strcmp(argv[i - 1], "-s") == 0)
+        {
+            seconds = 3;
+        }
+
+        if (strcmp(argv[i-1], "--time") == 0 ||
+            strcmp(argv[i - 1], "-t") == 0)
+        {
+            seconds = atoi(argv[i]);
+        }
+
+        if (strcmp(argv[i - 1], "--domain") == 0 ||
+            strcmp(argv[i - 1], "-d") == 0)
+        {
+            domain = atoi(argv[i]);
+        }
+    }
```

Comment:
So if it doesn't recognize an argument, it just ignores it? Shouldn't it complain and exit?